xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9_targets.h

```diff
@@ -67,18 +67,18 @@ struct s_rasterizer_target
 /* globals */
 
 extern IDirect3DTexture9* g_backbuffer_texture;
-extern IDirect3DSurface9* g_backbuffer_surface;
+// extern IDirect3DSurface9* g_backbuffer_surface;
```

Comment:
```cpp
struct s_rasterizer_data
{
	void** field_0;
	void** field_4;
	void** field_8;
	void** field_C;
	uint8 gap_10[4];
	void* bloom_globals;
	IDirect3D9* d3d;
	uint8 gap_1C[8];
	IDirect3DSurface9* d3d_surface_render_primary;
	IDirect3DSurface9* d3d_surface_render_primary_z;
	IDirect3DSurface9* d3d_surface_render_z_as_target_z;
	IDirect3DBaseTexture9* d3d_texture_render_resolved;
	IDirect3DSurface9* d3d_surface_render_resolved;
	IDirect3DSurface9* d3d_surface_screenshot;
	uint8 gap_3C[4];
	IDirect3DBaseTexture9* d3d_texture_backbuffer;
	IDirect3DSurface9* d3d_surface_backbuffer_depth;
	uint8 gap_48[8];
	IDirect3DBaseTexture9* d3d_texture_sun_glow_primary;
	IDirect3DSurface9* d3d_surface_sun_glow_primary;
	IDirect3DBaseTexture9* d3d_texture_sun_glow_secondary;
	IDirect3DSurface9* d3d_surface_sun_glow_secondary;
	IDirect3DBaseTexture9* d3d_texture_motion_sensor;
	IDirect3DSurface9* d3d_surface_motion_sensor;
	uint8 gap_68[4];
	IDirect3DSurface9* field_6C;
	IDirect3DSurface9* field_70;
	IDirect3DSurface9* field_74;
	IDirect3DSurface9* field_78;
	IDirect3DSurface9* field_7C;
	IDirect3DSurface9* field_80;
	IDirect3DSurface9* field_84;
	uint8 gap_88[4];
	void* d3d_device;
	uint8 gap_90[8];
};
ASSERT_STRUCT_SIZE(s_rasterizer_data, 152);

s_rasterizer_data* rasterizer_get_data()
{
    return Memory::GetAddress<s_rasterizer_data*>(0xA3C628);
}

rasterizer_get_data()->d3d_surface_render_primary;

```

you can also do:

```cpp
&rasterizer_get_data()->d3d_surface_render_primary;
```
and initialize the targets this way
```cpp
global_d3d_device->CreateRenderTarget(screen_bounds_width, screen_bounds_height, format, type, quality, false, &rasterizer_get_data()->d3d_surface_render_z_as_target_z, NULL);
```
and avoid all of these double pointer calls: 

```cpp
IDirect3DSurface9** global_d3d_surface_render_primary_get(void);
```

Reply:
basically a structure mapping game's memory

Reply:
Not a fan of doing this _at this moment_.
Could create confusion when referencing later as it'd be broken up into this struct and our globals, as well as any future render targets or additions we make.

Reply:
Personally not confused at all. I'm a big fan of doing this, exactly at this moment, compared to what is currently in.